opencloudApp/src/main/java/eu/opencloud/android/workers/UploadFileFromContentUriWorker.kt

```diff
@@ -252,14 +259,129 @@ class UploadFileFromContentUriWorker(
         val isChunkingAllowed = capabilitiesForAccount != null && capabilitiesForAccount.isChunkingAllowed()
         Timber.d("Chunking is allowed: %s, and file size is greater than the minimum chunk size: %s", isChunkingAllowed, fileSize > CHUNK_SIZE)
 
-        if (isChunkingAllowed && fileSize > CHUNK_SIZE) {
-            uploadChunkedFile(client)
+        // Prefer TUS for large files: optimistically try TUS and fall back on failure
+        val usedTus = if (fileSize > CHUNK_SIZE) {
+            Timber.d("Attempting TUS for large upload (size=%d, threshold=%d)", fileSize, CHUNK_SIZE)
+            val ok = try {
+                uploadTusFile(client)
+                true
+            } catch (e: Exception) {
+                Timber.w(e, "TUS flow failed, will fallback to existing upload methods")
+                false
+            }
+            Timber.d("TUS attempt result: %s", if (ok) "success" else "failed")
+            ok
         } else {
-            uploadPlainFile(client)
+            Timber.d("Skipping TUS: file too small (size=%d <= threshold=%d)", fileSize, CHUNK_SIZE)
+            false
+        }
+
+        if (!usedTus) {
+            if (isChunkingAllowed && fileSize > CHUNK_SIZE) {
```

Comment:
@TheOneRing says that actually the old chunking is being phased out. So basicaly, files are uploaded either with a single PUT or with TUS. So you can kick out this handling.